File: src/main/java/Data/Write.java
package Data;

import com.fasterxml.jackson.databind.ObjectMapper;
import Main.Inventory;
import Main.Main;
import main.InvSelector;

import java.io.File;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class Write {
    private static final Write instance = new Write();

    private Write(){}

    public static Write getInstance() {
        return instance;
    }

    private static void saveInPDF() {
        System.out.println("in PDF");
    }

    private static void saveInJSON() {
        System.out.println("Saving in JSON");
        ObjectMapper mapper = JsonUtil.getMapper();
        for(Inventory inv : InvSelector.INSTANCE.getInventoryMutableList()) {
            safeInJSON(mapper, inv);
        }
    }

    private static void safeInJSON(ObjectMapper mapper, Inventory inv) {
        try {
            Path path = Paths.get(System.getProperty("user.home"), ".DnD-Character-Manager", inv.getName() + ".json");
            File file = path.toFile();
            if(file.exists()) {
                System.out.println("File already exists at: " + file.getAbsolutePath());
                boolean delete = file.delete();
                System.out.println("Deleted: " + delete);
            }
            mapper.writeValue(new File(path.toString()), inv);
            System.out.println("Saved file");
        }
        catch (Exception e) {
            e.printStackTrace();
            System.out.println("could not save inv" + inv.getName() + " into file");
        }
    }

    public static void safe(Inventory inv) {
        if(Main.usePDF) System.out.println("saving in pdf is in development");
        else safeInJSON(JsonUtil.getMapper(), inv);
    }

    public static void removeInv(Inventory inv) {
        try {
            Path path = JsonUtil.getDataPath().resolve(inv.getName() + ".json");
            if(path.toFile().exists()) Files.delete(path);
        }
        catch (Exception e) {
            e.printStackTrace();
            System.out.println("could not delete inv" + inv.getName() + " from files");
        }
    }
}

File: src/main/kotlin/data/Write.kt
package data

import main.Inventory
import main.CharacterManager
import java.io.File
import java.nio.file.Files
import java.nio.file.Path

object Write {
    fun safe(inv: Inventory) {
        saveInJSON(inv)
    }

    private fun saveInJSON(inv: Inventory) {
        try {
            val path = JsonUtil.dataPath.resolve("${inv.name}.json")
            val file = path.toFile()
            if (file.exists()) {
                println("File already exists at: ${file.absolutePath}")
                val deleted = file.delete()
                println("Deleted: $deleted")
            }
            
            val jsonString = JsonUtil.json.encodeToString(Inventory.serializer(), inv)
            Files.writeString(path, jsonString)
            
            println("Saved file")
        } catch (e: Exception) {
            e.printStackTrace()
            println("could not save inv ${inv.name} into file")
        }
    }

    fun removeInv(inv: Inventory) {
        try {
            val path = JsonUtil.dataPath.resolve("${inv.name}.json")
            if (Files.exists(path)) {
                Files.delete(path)
            }
        } catch (e: Exception) {
            e.printStackTrace()
            println("could not delete inv ${inv.name} from files")
        }
    }
}
